--- RT3090_LinuxSTA_V2.4.0.4_20101217/os/linux/rt_linux.c.orig	2015-07-25 22:46:21.365160728 +0100
+++ RT3090_LinuxSTA_V2.4.0.4_20101217/os/linux/rt_linux.c	2015-07-25 22:46:44.702162479 +0100
@@ -1232,8 +1232,8 @@
 		pOSFSInfo->fsgid = current->fsgid;
 		current->fsuid = current->fsgid = 0;
 #else
-		pOSFSInfo->fsuid = current_fsuid();
-		pOSFSInfo->fsgid = current_fsgid();		
+		pOSFSInfo->fsuid = current_fsuid().val;
+		pOSFSInfo->fsgid = current_fsgid().val;
 #endif
 		pOSFSInfo->fs = get_fs();
 		set_fs(KERNEL_DS);
--- RT3090_LinuxSTA_V2.4.0.4_20101217/os/linux/pci_main_dev.c.orig	2015-07-25 22:57:58.445213032 +0100
+++ RT3090_LinuxSTA_V2.4.0.4_20101217/os/linux/pci_main_dev.c	2015-07-25 22:58:57.471217461 +0100
@@ -28,6 +28,12 @@
 #include "rt_config.h"
 #include <linux/pci.h>
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
+#define __devexit
+#define __devinit
+#define __devinitdata
+#endif
+
 //
 // Function declarations
 //
@@ -86,10 +92,14 @@
     name:       "rt2860",
     id_table:   rt2860_pci_tbl,
     probe:      rt2860_probe,
-#if LINUX_VERSION_CODE >= 0x20412
-    remove:     __devexit_p(rt2860_remove_one),
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3,8,0)
+remove:     rt2860_remove_one,
 #else
-    remove:     __devexit(rt2860_remove_one),
+#if LINUX_VERSION_CODE >= 0x20412 
+ remove:     __devexit_p(rt2860_remove_one), 
+#else 
+ remove:     __devexit(rt2860_remove_one), 
+#endif 
 #endif
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,0)
